freud/density.pyx

```diff
@@ -732,3 +733,126 @@ cdef class RDF(_SpatialHistogram1D):
+
+    where :math:`k` is the number of neighbors a particle has, :math:`V_i` is
+    the volume of the pyramid (or area of the triangle) whose base is the
+    Voronoi polytope facet between the central particle and neighobor :math:`i`
+    and whose height is half the distance vector, and :math:`V` is the
+    volume/area of the Voronoi polytope.
+
+    Args:
+        powers (list[float]): The powers to compute the continuous coordination
+            number for.
+        compute_log (`bool`, optional): Whether to compute the log continuous
+            coordination number.
+        compute_exp (`bool`, optional): Whether to compute the exp continuous
+            coordination number.
+    """
+    cdef freud._density.ContinuousCoordination* thisptr
+
+    def __cinit__(self, powers, compute_log=False, compute_exp=False):
+        self.thisptr = new freud._density.ContinuousCoordination(
+            powers, compute_log, compute_exp)
+
+    def __dealloc__(self):
+        del self.thisptr
+
+    def compute(self, system, voronoi=None):
+        r"""Calculates the local density for the specified points.
+
+        Example::
+
+            >>> import freud
+            >>> box, points = freud.data.make_random_system(10, 100, seed=0)
+            >>> # Compute ContinuousCoordination
+            >>> coord = freud.density.ContinuousCoordination([2, 4], True)
+            >>> coord.compute(system=(box, points))
+            freud.density.ContinuousCoordination(...)
+
+        Args:
+            system:
+                Any object that is a valid argument to
+                :class:`freud.locality.NeighborQuery.from_system`.
+            voronoi:
+                A precomputed Voronoi compute object. If provided, the object is
+                assumed to have already been computed for the passed in system.
+                (Default value: None).
+        """  # noqa E501
+        cdef freud.locality.Voronoi cpp_voronoi
+        cdef freud.locality.NeighborList cpp_nlist
+        cdef bool is2D
+
+        neighbor_query = freud.locality.NeighborQuery.from_system(system)
+        is2D = neighbor_query.box.is2D
+
+        if voronoi is None:
+            voronoi = freud.locality.Voronoi()
+            voronoi.compute(system)
+        cpp_voronoi, cpp_nlist = (voronoi, voronoi.nlist)
+        self.thisptr.compute(cpp_voronoi.thisptr, cpp_nlist.get_ptr(), is2D)
+        return self
+
+    @_Compute._computed_property
+    def coordination(self):
+        """(:math:`(N_{points}, N_{coord}`) :class:`numpy.ndarray`: \
+                coordination of points per query point."""
+        return freud.util.make_managed_numpy_array(
+            &self.thisptr.getCoordination(),
+            freud.util.arr_type_t.FLOAT)
+
+    @property
+    def powers(self):
+        """list[float]: The powers to compute the continuous coordination number.
+
+        Changes to this property are not reflected when computing coordination
+        numbers.
```

Comment:
Not necessarily. Operations like `compute.powers.append(4)` would be completely syntactically correct and work... just not as intended by a user. We could mitigate this by manual converting to a tuple to prevent these shenanigans.

Reply:
For other cases like this, we have ignored that fact that a user may try to do that. It hasn't led to issues that I'm aware of.

Reply:
So do you want me to remove the comment or just not bother with changing it to a tuple?

Reply:
Everything can be left as-is.